.github/copilot-instructions.md

```diff
@@ -1,21 +1,120 @@
+- Format code: `pnpm format`
+- Run linting: `pnpm lint` (~1 minute)
+- Fix lint issues: `pnpm lint:fix`
+
+### Essential TypeSpec Development Workflow
+
+1. **ALWAYS** run the full build process after repository clone: `pnpm install && pnpm build`
+2. Start watch mode: `pnpm watch`
+3. Test TypeSpec compilation works:
+
+   ```bash
+   # Create test project
+   mkdir test-tsp && cd test-tsp
+   echo 'import "@typespec/rest"; import "@typespec/openapi3"; op ping(): void;' > main.tsp
+   echo '{"dependencies": {"@typespec/compiler": "latest", "@typespec/rest": "latest", "@typespec/openapi3": "latest"}}' > package.json
+   
+   # Install and compile
+   /path/to/typespec/packages/compiler/cmd/tsp.js install
+   /path/to/typespec/packages/compiler/cmd/tsp.js compile main.tsp --emit @typespec/openapi3
+   ```
+
+4. Always format and lint before completing changes: `pnpm format && pnpm lint:fix`
+
+## Repository Structure
+
+### Key Packages (packages/)
+
+- **compiler**: Core TypeSpec compiler and CLI tool
+- **http, rest, openapi3**: Standard HTTP/REST API libraries
+- **versioning**: API versioning support
+- **json-schema**: JSON Schema emitter
+- **prettier-plugin-typespec**: Code formatting support
+- **typespec-vscode, typespec-vs**: Editor extensions
+- **playground**: Interactive TypeSpec playground
+- **website**: Documentation website (typespec.io)
+
+### Important Directories
+
+- `/packages/`: All TypeSpec packages and libraries
+- `/e2e/`: End-to-end integration tests
+- `/website/`: Documentation website source
+- `/eng/`: Build engineering and automation scripts
+- `/.github/workflows/`: CI/CD pipeline definitions
+
+## Manual Validation After Changes
+
+**ALWAYS perform these validation steps after making changes:**
+
+1. **Basic functionality test**: Create and compile a simple TypeSpec file as shown above
+2. **Build validation**: Run full build to ensure no build breaks: `pnpm build`
+3. **Test validation**: Run relevant tests: `pnpm test`
+4. **Code quality**: Ensure formatting and linting pass: `pnpm format:check && pnpm lint`
+
+## Website Development
+
+- Navigate to website: `cd website`
+- Start development server: `pnpm start` (runs on port 4321)
+- Build website: `pnpm build`
+- The website includes documentation, API references, and the playground
+
+## Critical Timing and Performance Notes
+
+- **NEVER CANCEL** long-running commands - builds can take 7+ minutes, tests 5+ minutes
+- Set explicit timeouts: Build commands need 15+ minutes, test commands need 10+ minutes
+- Package installation: ~1.5 minutes
+- Full rebuild from clean state: ~7 minutes
+- Full test suite: ~5 minutes
+- Lint check: ~1 minute
+- E2E tests: ~1 minute
+
+## Common Development Tasks
+
+- Add change description: `pnpm change add`
+- Generate external signatures: `pnpm gen-compiler-extern-signature`
+- Regenerate samples: `pnpm regen-samples`
+- Regenerate docs: `pnpm regen-docs`
+- Package all for release: `pnpm pack:all`
```

Comment:
I'm not sure packaging would be a common task since we use workspaces/use other tooling for preparing releases.